File: kubernetes-client-openapi-informer/src/test/java/io/micronaut/kubernetes/client/openapi/informer/example1/CustomObject.java
package io.micronaut.kubernetes.client.openapi.informer.example1;

import io.micronaut.kubernetes.client.openapi.common.KubernetesObject;
import io.micronaut.kubernetes.client.openapi.model.V1ObjectMeta;
import io.micronaut.serde.annotation.Serdeable;

//tag::get[]
@Serdeable
public class CustomObject implements KubernetesObject {

    private String apiVersion;

    private String kind;

    private V1ObjectMeta metadata;

    private String value;

    // getters/setters omitted
    //end::get[]

    @Override
    public String getApiVersion() {
        return apiVersion;
    }

    public void setApiVersion(String apiVersion) {
        this.apiVersion = apiVersion;
    }

    @Override
    public String getKind() {
        return kind;
    }

    public void setKind(String kind) {
        this.kind = kind;
    }

    @Override
    public V1ObjectMeta getMetadata() {
        return metadata;
    }

    public void setMetadata(V1ObjectMeta metadata) {
        this.metadata = metadata;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
//tag::get[]
}
//end::get[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/openapi/informer/example1/CustomObject.kt
package io.micronaut.docs.openapi.informer.example1

import io.micronaut.kubernetes.client.openapi.common.KubernetesObject
import io.micronaut.kubernetes.client.openapi.model.V1ObjectMeta
import io.micronaut.serde.annotation.Serdeable

//tag::get[]
@Serdeable
class CustomObject : KubernetesObject {

    private var apiVersion: String? = null

    private var kind: String? = null

    private var metadata: V1ObjectMeta? = null

    var value: String? = null

    // getters/setters omitted
    //end::get[]

    override fun getApiVersion(): String? = apiVersion

    fun setApiVersion(apiVersion: String?) {
        this.apiVersion = apiVersion
    }

    override fun getKind(): String? = kind

    fun setKind(kind: String?) {
        this.kind = kind
    }

    override fun getMetadata(): V1ObjectMeta? = metadata

    fun setMetadata(metadata: V1ObjectMeta?) {
        this.metadata = metadata
    }
//tag::get[]
}
//end::get[]
